### ISSUE TITLE
To add LCP_ECHOINTERVAL/LCP_MAXECHOFAILS configuration items (IDFGH-3098)

### ISSUE THREAD

**Issue by AxelLin** [espressif/esp-idf#5119]
Hi @projectgus 

Base on the discussion on this thread:
https://www.esp32.com/viewtopic.php?f=13&t=13756

Please add below configuration items to PPP settings via menuconfig

LCP_ECHOINTERVAL
LCP_MAXECHOFAILS

Thanks,
Axel

**Comment by david-cermak** [comment]
@AxelLin 
Thanks for posting this feature request, we will look into it, but this may take some time, since we'd like to test at least on general level

Meantime, might I suggest enabling this options simply on project level?

The below code in your project CMake file should apply the define to the lwip component
```
idf_component_get_property(lwip lwip COMPONENT_LIB)

target_compile_definitions(${lwip} PRIVATE "-DLCP_ECHOINTERVAL=XXX")
```

**Comment by AxelLin** [comment]
> Meantime, might I suggest enabling this options simply on project level?
> 
> The below code in your project CMake file should apply the define to the lwip component
> 
> ```
> idf_component_get_property(lwip lwip COMPONENT_LIB)
> 
> target_compile_definitions(${lwip} PRIVATE "-DLCP_ECHOINTERVAL=XXX")
> ```
Hi @david-cermak 
Above change works.
Thanks.

**Comment by projectgus** [comment]
@AxelLin Glad that the workaround is functioning, but we can leave this request open until we actually have this feature in ESP-IDF.

**Comment by JosueSanchezEZM** [comment]
This workaround does not work for me: 
![image](https://user-images.githubusercontent.com/76057615/108369128-3cf53480-71c9-11eb-9ee4-9b0040cc1938.png)

Makefile:
![image](https://user-images.githubusercontent.com/76057615/108369264-657d2e80-71c9-11eb-9bb6-e0bfb280648e.png)

**Comment by JosueSanchezEZM** [comment]
Is this functionality already available in esp-idf 4.3/release ?
